packages/theme/src/cli/services/check.ts

```diff
@@ -0,0 +1,359 @@
+    // Format the file path to be relative to the theme root.
+    // Remove the leading slash agnostic of windows or unix.
+    const headlineFilePath = filePath.replace(themeRootPath, '').slice(1)
+
+    renderInfo({
+      headline: headlineFilePath,
+      body: formatOffenses(offensesByFile[filePath]!),
+    })
+  })
+}
+
+export function formatOffensesJson(offensesByFile: OffenseMap): TransformedOffenseMap[] {
+  return Object.entries(offensesByFile).map(([path, offenses]) => {
+    let errorCount = 0
+    let suggestionCount = 0
+    let styleCount = 0
+
+    const transformedOffenses = offenses.map((offense: Offense) => {
+      if (offense.severity === Severity.ERROR) errorCount++
+      if (offense.severity === Severity.WARNING) suggestionCount++
+      if (offense.severity === Severity.INFO) styleCount++
+
+      return {
+        check: offense.check,
+        severity: offense.severity,
+        start_row: offense.start.line,
+        start_column: offense.start.character,
+        end_row: offense.end.line,
+        end_column: offense.end.character,
+        message: offense.message,
+      }
+    })
+
+    return {
+      path,
+      offenses: transformedOffenses,
+      errorCount,
+      suggestionCount,
+      styleCount,
+    }
+  })
+}
+
+/**
+ * Handles the process exit based on the offenses and fail level.
+ */
+export function handleExit(offenses: Offense[], failLevel?: FailLevel) {
+  // If there is no fail level set, exit with 0
+  if (!failLevel) process.exit(0)
+
+  const failSeverity = failLevelToSeverity(failLevel)
+  const shouldFail = offenses.some((offense) => offense.severity < failSeverity)
+
+  process.exit(shouldFail ? 1 : 0)
+}
+
+/**
+ * Adds a '#' character at the start of each line in a string.
+ */
+function commentString(input: string): string {
+  return input
+    .split('\n')
+    .map((line) => `# ${line}`)
+    .join('\n')
+}
+
+export async function initConfig(root: string) {
+  const basefile = '.theme-check.yml'
+  const filePath = joinPath(root, basefile)
+  if (await fileExists(filePath)) {
+    outputInfo(`${basefile} already exists at ${root}`)
+    return
+  }
+
+  // The initialized config will extend the recommended settings and
+  // will simply show the commented checks for the user to customize.
+  const {settings} = await loadConfig(undefined, root)
+  const checksYml = commentString(YAML.stringify(settings))
+
+  const initConfigYml = YAML.stringify({extends: 'theme-check:recommended', ignore: ['node_modules/**']})
+
+  await writeFile(filePath, `${initConfigYml}${checksYml}`)
+
+  outputSuccess(`Created ${basefile} at ${root}`)
+}
+
+export async function runThemeCheck(themeRoot: string, configPath?: string) {
+  let themeCheckResults = {} as ThemeCheckRun
+
+  const themeCheckTask: Task = {
+    title: `Performing theme check. Please wait...\nEvaluating ${themeRoot}`,
+    task: async () => {
+      themeCheckResults = await themeCheckRun(themeRoot, configPath)
+    },
+  }
```

Comment:
That's the thing that might need an option if `--output json` is selected. This pollutes STDOUT. Logging should happen on STDERR!

Reply:
Very good catch here. I think we probably need to drop `renderTask` in this case.